packages/client/src/registry/NodeManager.ts

```diff
@@ -116,12 +100,42 @@ export class NodeManager {
 
 		// This function is throttled to run at most once every minute.
 		// Also, we're able to get the lastUrlList$ value instantly, as soon as the first heartbeat message arrives.
-		this.throttledUpdateList = throttle(
-			() => this.updateNodeUrlList(),
-			60_000,
-			{ leading: true }
+		// this.throttledUpdateList = throttle(
+		// 	() => this.updateNodeUrlList(),
+		// 	60_000,
+		// 	{ leading: true }
+		// );
+		// this.throttledUpdateList();
+	}
+
+	public async getRandomNodeUrl() {
+		const nodeAddresses = shuffle(
+			await queryAllReadonlyContracts(
+				(contract: LogStoreNodeManagerContract) => {
+					return contract.nodeAddresses();
+				},
+				this.logStoreManagerContractsReadonly
+			)
 		);
-		this.throttledUpdateList();
+
+		for (const nodeAddress of nodeAddresses) {
+			const node = await queryAllReadonlyContracts(
+				(contract: LogStoreNodeManagerContract) => {
+					return contract.nodes(nodeAddress);
+				},
+				this.logStoreManagerContractsReadonly
+			);
+			if (node.metadata.includes('http')) {
+				try {
+					const metadata = JSON.parse(node.metadata) as NodeMetadata;
+					return metadata.http;
+				} catch (e) {
+					// ...
+				}
+			}
+		}
+
+		throw new Error('There are no nodes with a proper metadata');
 	}
 
 	private getNodeFromAddress = async (address: string) => {
```

Comment:
I assume this is commented out because we've reverted to random network node selection?

Reply:
Yes, it is. I'd like to keep it here commented until we deploy a new release to the Alpha Network, and ensure that it's ok to go without it.